src/main/java/org/opentripplanner/middleware/tripmonitor/TrustedCompanion.java

```diff
@@ -0,0 +1,213 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.opentripplanner.middleware.tripmonitor.jobs.CheckMonitoredTrip.SETTINGS_PATH;
+import static org.opentripplanner.middleware.utils.I18nUtils.label;
+
+public class TrustedCompanion {
+
+    private TrustedCompanion() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    private static final String AWS_API_SERVER = ConfigUtils.getConfigPropertyAsText("AWS_API_SERVER");
+    private static final String AWS_API_STAGE = ConfigUtils.getConfigPropertyAsText("AWS_API_STAGE");
+    private static final String OTP_UI_URL = ConfigUtils.getConfigPropertyAsText("OTP_UI_URL");
+    private static final String TRUSTED_COMPANION_CONFIRMATION_PAGE_URL = ConfigUtils.getConfigPropertyAsText("TRUSTED_COMPANION_CONFIRMATION_PAGE_URL");
+    public static final String ACCEPT_KEY = "acceptKey";
+    public static final String EMAIL_FIELD_NAME = "email";
+
+    /** Note: This path is excluded from security checks, see {@link OtpMiddlewareMain#initializeHttpEndpoints()}. */
+    public static final String ACCEPT_DEPENDENT_PATH = "api/secure/user/acceptdependent";
+
+    /**
+     * Accept a request from another user to be their dependent. This will include both companions and observers. If
+     * successful redirect the user to the confirmation page, else redirect to the error page with related error.
+     */
+    public static OtpUser acceptDependent(Request request, Response response) {
+        try {
+            String acceptKey = getAcceptKeyFromRequest(request);
+            OtpUser dependentUser = getUserFromAcceptKey(acceptKey);
+            OtpUser relatedUser = getRelatedUserFromEmail(dependentUser, acceptKey);
+
+            if (relatedUser != null) {
+                Optional<RelatedUser> relatedUserToUpdate = dependentUser.relatedUsers
+                    .stream()
+                    .filter(related -> related.email.equals(relatedUser.email))
+                    .findFirst();
+                relatedUserToUpdate.ifPresent(value -> value.status = RelatedUser.RelatedUserStatus.CONFIRMED);
+
+                // Maintain a list of dependents.
+                relatedUser.dependents.add(dependentUser.id);
+                Persistence.otpUsers.replace(relatedUser.id, relatedUser);
+                // Update list of related users.
+                Persistence.otpUsers.replace(dependentUser.id, dependentUser);
+
+                // Redirect to confirmation page and provide dependent user information.
+                response.redirect(TRUSTED_COMPANION_CONFIRMATION_PAGE_URL);
+                return dependentUser;
+            }
+        } catch (IllegalArgumentException e) {
+            response.redirect(
+                String.format("%s?error=%s", TRUSTED_COMPANION_CONFIRMATION_PAGE_URL, Message.ACCEPT_DEPENDENT_ERROR.get(null))
```

Comment:
The message should be encoded.

Reply:
@binh-dam-ibigroup This needs to be the related user's locale which we are unlikely to have at this point. To save pushing this to a subsequent PR, shall I update the accept link to include the locale?

Reply:
@br648 The target user locale is indeed not directly known but can be looked up (because an account exists), so sure you can add that.

Reply:
Update in: https://github.com/ibi-group/otp-middleware/pull/256/commits/99341cb2a25e6633d69fb424482c7936fb6adccc